src/components/ChatEntry.js

```diff
@@ -1,22 +1,45 @@
 import React from 'react';
 import './ChatEntry.css';
+
+import TimeStamp from './TimeStamp';
 import PropTypes from 'prop-types';
 
 const ChatEntry = (props) => {
+  const changeHeart = () => {
+    props.toggleHeartCallback(props.id);
+  };
+
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div
+      className={`chat-entry ${
+        props.sender === 'Estragon' ? 'remote' : 'local'
+      }`}
+    >
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{props.body}</p>
+        <p className="entry-time">
+          <TimeStamp time={props.timeStamp} />
+        </p>
+        <button className="like" onClick={changeHeart}>
+          {props.liked ? '❤️' : '🤍'}
+        </button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  liked: PropTypes.bool,
+  toggleHeartCallback: PropTypes.func,
+};
+
+ChatEntry.defaultProps = {
+  toggleHeartCallback: () => {},
```

Comment:
Ooh! I actually did not know of this functionality of React. I was about to leave a comment that I think `toggleHeartCallback` should be `.isRequired`, but this is a great way to make that unnecessary, and without breaking the tests (who do not set this prop) either!

Reply:
Though, looking at `ChatLog`, I would have also did the same thing there. `PropTypes` is a great tool to catch a lot of basic bugs, so don't hesitate to use it!